nf_core/sync.py

```diff
@@ -333,53 +318,128 @@ def update_existing_pull_request(self, pr_title, pr_body_text):
             if r.status_code == 200:
                 log.debug("GitHub API PR-update worked:\n{}".format(r_pp))
-                log.info("Updated GitHub PR: {}".format(r_json["html_url"]))
+                log.info("Closed GitHub PR: {}".format(r_json["html_url"]))
                 return True
             # Something went wrong
             else:
-                log.warning("Could not update PR ('{}'):\n{}\n{}".format(r.status_code, pr_update_api_url, r_pp))
+                log.warning(f"Could not close PR ('{r.status_code}'):\n{pr_update_api_url}\n{r_pp}")
                 return False
 
         # Something went wrong
         else:
             log.warning("Could not list open PRs ('{}')\n{}\n{}".format(r.status_code, list_prs_url, r_pp))
-            return False
+
+    def create_merge_base_branch(self):
+        """Create a new branch from the updated TEMPLATE branch
+        This branch will then be used to create the PR
+        """
+        # Check if branch exists already
+        branch_list = [b.name for b in self.repo.branches]
+        if self.merge_branch in branch_list:
+            original_merge_branch = self.merge_branch
+            # Try to create new branch with number at the end
+            # If <branch_name>-2 already exists, increase the number until branch is new
+            branch_no = 2
+            self.merge_branch = original_merge_branch + "-" + str(branch_no)
+            while self.merge_branch in branch_list:
+                branch_no += 1
+                self.merge_branch = original_merge_branch + "-" + str(branch_no)
+            log.info(
+                "Branch already existed: '{}', creating branch '{}' instead.".format(
+                    original_merge_branch, self.merge_branch
+                )
+            )
+
+        # Create new branch and checkout
+        log.info("Checking out merge base branch {}".format(self.merge_branch))
+        try:
+            self.repo.create_head(self.merge_branch)
+        except git.exc.GitCommandError as e:
+            raise SyncException(f"Could not create new branch '{self.merge_branch}'\n{e}")
+
+    def push_merge_branch(self):
+        """Push the newly created merge branch to the remote repository"""
+        log.info("Pushing {} branch to remote".format(self.merge_branch))
+        try:
+            origin = self.repo.remote()
+            origin.push(self.merge_branch)
+        except git.exc.GitCommandError as e:
+            raise PullRequestException(f"Could not push branch '{self.merge_branch}':\n  {e}")
+
+    def make_pull_request(self):
+        """Create a pull request to a base branch (default: dev),
+        from a head branch (default: TEMPLATE)
+
+        Returns: An instance of class requests.Response
+        """
+        # Check that we know the github username and repo name
+        if self.gh_username is None and self.gh_repo is None:
+            raise PullRequestException("Could not find GitHub username and repo name")
```

Comment:
I wonder if we also just check for this once before we start all the different PR function calls. Might be nice to clean it out of these functions.